src/tools/pub/dependencies.md

```diff
@@ -70,62 +58,58 @@ Pub can use the following sources to locate packages:
 * [Git packages](#git-packages)
 * [Path packages](#path-packages)
 
-### SDK
-
-The SDK source is used for any SDKs that are shipped along with packages,
-which may themselves be dependencies.
-Currently, Flutter is the only SDK that is supported.
-
-The syntax looks like this:
 
-{% prettify yaml tag=pre+code %}
-dependencies:
-  flutter_driver:
-    sdk: flutter
-    version: ^0.0.1
-{% endprettify %}
-
-The identifier after `sdk:` indicates which SDK the package comes from.
-If it's `flutter`, the dependency is satisfiable as long as:
-
-* Pub is running in the context of the `flutter` executable
-* The Flutter SDK contains a package with the given name
-* That package's version matches the version constraint
-
-If it's an unknown identifier, the dependency is always considered unsatisfied.
-
-A package with an `sdk` dependency
-must have a Dart SDK constraint with a minimum version of at least 1.19.0.
-This constraint ensures that older versions of pub won't
-install packages that have `sdk` dependencies.
 
 ### Hosted packages
 
 A *hosted* package is one that can be downloaded from the pub.dev site
 (or another HTTP server that speaks the same API). Here's an example
 of declaring a dependency on a hosted package:
 
-{% prettify yaml tag=pre+code %}
+```yaml
 dependencies:
   transmogrify: ^1.4.0
-{% endprettify %}
+```
 
 This example specifies that your package depends on a hosted package named
-`transmogrify` and will work with any version from 1.4.0 to 2.0.0
+`transmogrify` and works with any version from 1.4.0 to 2.0.0
 (but not 2.0.0 itself).
 
-If you want to use your own package server, you can use a description that
-specifies its URL:
+If you want to use your own package repository,
+you can use `hosted` to specify its URL.
+The following YAML code creates a dependency on the `transmogrify` package
+using the `hosted` source.
 
 {% prettify yaml tag=pre+code %}
+environment: 
+  sdk: >=[!2.15.0!] < 3.0.0
+
 dependencies:
   transmogrify:
-    hosted:
-      name: transmogrify
-      url: http://your-package-server.com
+    [!hosted: http://some-package-server.com!]
```

Comment:
With the new `https://dart.dev/tools/linter-rules#secure_pubspec_urls` linter rule, perhaps we should use `https` links in the examples.

```suggestion
    [!hosted: https://some-package-server.com!]
```